art/attacks/evasion/hop_skip_jump.py

```diff
@@ -140,7 +150,14 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
         x_adv_init = kwargs.get("x_adv_init")
 
         if x_adv_init is not None:
+            # Add mask param to the x_adv_init
+            if mask is not None:
+                for i in range(x.shape[0]):
+                    x_adv_init[i] = x_adv_init[i] * mask[i] + x[i] * (1 - mask[i])
```

Comment:
Is this loop necessary? Could it be replace by a vectorised operation?

Reply:
Yes it is mandatory. If shapes of x and mask are different, then broadcasting will give error. For example x shape is (10, 32, 32, 3) and mask shape is (10, 32, 3), the vectorised version will throw error.